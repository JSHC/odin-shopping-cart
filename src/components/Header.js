import { Link } from "react-router-dom";
import "../styles/Header.css";
import ShoppingCartLink from "./ShoppingCartLink";

const Header = (props) => {
    const { cartItems } = props;
    const getNumberOfCartItems = () => {
        if (!cartItems || cartItems.length === 0) {
            return 0;
        } else {
            const totalQuantity = cartItems
                .map((item) => item.quantity)
                .reduce((prev, curr) => {
                    return prev + curr;
                });

            return totalQuantity;
        }
    };

    return (
        <div className="header">
            <nav>
                <ul className="header-nav-list">
                    <li className="header-logo">
                        <Link to="/">JAFS</Link>
                    </li>
                    <ul className="header-links">
                        <li>
                            <Link to="/">Home</Link>
                        </li>
                        <li>
                            <Link to="/shop">Shop</Link>
                        </li>
                    </ul>
                    <li>
                        <ShoppingCartLink
                            cartItemsNumber={getNumberOfCartItems()}
                        />
                    </li>
                </ul>
            </nav>
        </div>
    );
};

export default Header;
