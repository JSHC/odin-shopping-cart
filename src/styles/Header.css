.header {
    background-color: #222;
    grid-area: header;
    color: #fff;
}

.header nav ul {
    list-style: none;
    display: flex;
    justify-content: space-around;
}

.header a {
    color: #fff;
    text-decoration: none;
}