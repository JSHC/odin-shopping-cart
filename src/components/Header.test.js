import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from './Header';

describe('Header component', () => {
    it('renders', () => {
        const { container } = render(<Header />);
        expect(container).toMatchSnapshot();
    }) 
})