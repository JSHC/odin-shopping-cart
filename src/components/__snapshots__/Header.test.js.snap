// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders 1`] = `
<div>
  <div
    class="header"
  >
    <nav>
      <ul>
        <li>
          Home
        </li>
        <li>
          Shopping Cart
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`Header component renders nav list 1`] = `
<div>
  <div
    class="header"
  >
    <nav>
      <ul>
        <li>
          Home
        </li>
        <li>
          Shopping Cart
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
